unlock-app/src/hooks/usePaywallLocks.ts

```diff
@@ -17,8 +20,18 @@ export const usePaywallLocks = (lockAddresses: string[]) => {
     })
 
     const locks = await Promise.all(lockPromises)
-    // getLock doesn't always return the lock address apparently
-    locks.forEach((lock, index) => (lock.address = lockAddresses[index]))
+
+    locks.forEach((lock, index) => {
+      if (lock.currencyContractAddress) {
+        // This is the easiest place to determine which tokens we need
+        // to query for the user's balance of. The values are tracked
+        // in the useGetTokenBalance hook.
```

Comment:
Unfortunately, this is a bit at-odds with the ideal of showing locks before we have an account. This is fine for now and we can figure out how to resolve it in the future.